### PR TITLE
tmux: Add caveat with workaround for new default terminal

### PR DESCRIPTION
**Pull request by utkonos**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This is a workaround for #102748

With the release of tmux 3.3, the default terminal chaged to `tmux-256color` which is not always available without making some changes (please look at the thread in the issue above for more information. Because tmux is non-functional without one among the different fixes and workarounds listed in that issue, it would be good to at minumum add a caveat to help someone understand why tmux is non-functional after the most recent upgrade, and a quick fix to return to the terminal used in previous versions.

### CODE REVIEW

**Review by carlocab — COMMENTED** [review]
This part is likely only relevant to macOS (Linux typically provides `tmux-256color`), so we should omit the added caveat on non-macOS systems.

We probably also want to make it much shorter (most people don't read `caveats`, especially if they're more than a couple of lines).

Also, why not just add
```
set -g default-terminal screen-256color
```
to `etc/"tmux.conf"` (on macOS)?

**Comment by utkonos** [comment]
@carlocab Thanks for the feedback. I have decreased the verbosity. Also, I re-read the Formula Cookbook and then grepped it for `linux` and `macos` and it is not clear how to omit a caveat for one OS or to specify an OS for a caveat. Can you point me to how to do that?

File: Formula/tmux.rb [hunk 1]
```diff
@@ -67,6 +67,9 @@ def caveats
     <<~EOS
       Example configuration has been installed to:
         #{opt_pkgshare}
+
+      On macOS, add the following to etc/tmux.conf or ~/.tmux.conf
+        set -g default-terminal screen-256color
```
Comment (carlocab) [inline Formula/tmux.rb:72]: I don't mean that we should suggest adding the path `etc/tmux.conf` to the caveat -- I meant that we should consider creating a file `tmux.conf` in `$HOMEBREW_PREFIX/etc` (which is the path given by the variable `etc`) that contains only
```tmux
set -g default-terminal screen-256color
```
(but only on macOS).

I was wondering what the reasons to prefer one approach over the other would be.

**Comment by carlocab** [comment]
Something like this might work:
```ruby
def caveats
    s = <<~EOS
      Example configuration has been installed to:
        #{opt_pkgshare}
    EOS

    s += if OS.mac?
      <<~EOS
        If you have issues with the default `TERM` of `tmux-256color`,
        it may help to add the following to your `~/.tmux.conf`:
          set -g default-terminal screen-256color
      EOS
    end

    s
  end
```

File: Formula/tmux.rb (hunk 1 above)
Comment (utkonos) [reply Formula/tmux.rb:72]: The only reason to prefer the caveat is that I don't know how to create and populate a file correctly in homebrewese. I've re-read the Formula Cookbook and I don't see an example of creating a file like this. I know how to do it using shell and redirection, but I feel like that is not the right way to do it. I'm trying to pick through https://rubydoc.brew.sh/ to see if I can identify a good option. I check a bunch of Formula at random and didn't see any that do something like this.

Comment (SMillerDev) [reply Formula/tmux.rb:72]:
```rb
(etc/"file").write <<~EOS 
  sometext
EOS
```

File: Formula/tmux.rb [hunk 2]
```diff
@@ -61,6 +61,12 @@ def install
 
     pkgshare.install "example_tmux.conf"
     bash_completion.install resource("completion")
+
+    if OS.mac?
+      (etc/"file").write <<~EOS
```
Comment (SMillerDev) [inline Formula/tmux.rb:66]: This will still need the correct filename

**Comment by SMillerDev** [comment]
I moved the block to `post_install` so we can check if people already have configuration there

**Comment by utkonos** [comment]
Good idea. Don't want to stomp someone's config. Maybe I'm overthinking scenarios, but if someone has a config and there is silently no change and they don't have this line in the existing config, they will encounter the behavior this is seeking to fix anyway.

Does the current state of the fix still deserve a caveat?

**Comment by SMillerDev** [comment]
I think people who made a custom config will know how to fix this.

**Comment by owine** [comment]
FYI - opened #103311 for a version update

**Review by SMillerDev — DISMISSED** [review]

**Comment by utkonos** [comment]
The version update skipped the line. I pulled that into the PR branch. CI checks running now.

**Comment by SMillerDev** [comment]
Could you rebase the branch? Otherwise the CI will have trouble merging.

**Comment by SMillerDev** [comment]
With 400+ files changed, I don't think that rebase worked as expected.

**Comment by utkonos** [comment]
Yes, sorry. That didn't work. I'll create a new PR and close this one. I think that's going to be faster than sorting this out.

**Pull request closed without merging**